test/IRGen/existentials_opaque_boxed.sil

```diff
@@ -140,8 +140,8 @@ entry:
 // CHECK-32:[[ALIGNMASK:%.*]] = and i32 [[FLAGS]], 255
 // CHECK-16:[[T0:%.*]] = trunc i32 [[FLAGS]] to i16
 // CHECK-16:[[ALIGNMASK:%.*]] = and i16 [[T0]], 255
-// CHECK:   [[HEADERSIZEPLUSALIGN:%.*]] = add {{(i64 16|i32 8)}}, [[ALIGNMASK]]
-// CHECK:   [[NOTALIGNMASK:%.*]] = xor {{(i64|i32)}} [[ALIGNMASK]], -1
+// CHECK-DAG:   [[HEADERSIZEPLUSALIGN:%.*]] = add {{(i64 16|i32 8)}}, [[ALIGNMASK]]
+// CHECK-DAG:   [[NOTALIGNMASK:%.*]] = xor {{(i64|i32)}} [[ALIGNMASK]], -1
```

Comment:
What's with these CHECK-DAG lines?  Why is this different on MSVC?

Reply:
I don't know why, but when targeting windows, the order of the instructions is swapped.  Its perfectly valid IR (there are no dependencies between the instructions and either order is totally valid).  That's why I wanted to get another pair of eyes on this change :-).

Reply:
Is it a host compiler difference rather than a target difference *per se*?  It's easy to imagine that we're relying on argument order somewhere, like if someone's written `Builder.CreateAnd(Builder.CreateAdd(...), Builder.CreateNot(...))`.

Reply:
Hmm, thats a good point.  This was building with `clang-cl` though, so the underlying compiler should be the same.  However, you are right in that the order of evaluation for that is different (windows goes right to left, Linux/macOS goes left to right).

Reply:
Right, for some reason `clang` decided that using a different order of evaluation was too breaking for `clang-cl`.

Reply:
Unfortunately, the argument evaluation ordering is something that code does sorta depend on (clearly, we do ;-)).  So, it seems like a reasonable choice on that part for maintaining compatibility with the dominant compiler on that platform.

Reply:
mmm, I'm not sure if I have a good way to do that, and definitely not to find all the possible instances.

Reply:
All I'm asking is for you to fix the places leading to this instead of fixing the test.  This should be quite easy to find because they're executed in response to a specific SIL instruction sequence.

HINT: at least one of them is GenExistential.cpp:2111. :)

Reply:
Yeah, I had that one already, but, fixing the two in GenExistentials doesnt really provide a good way to catch regressions or to ensure that it doesn't happen elsewhere.  I'm definitely seeing other ordering issues in the IRGen tests.

Reply:
Changes in the test output, on the other hand, are a great way to find places this is happening today. It'd be better to fix these tests by eliminating the argument evaluation order dependencies, which could be serious bugs.

Reply:
Exactly.  These need to be fixed; I'm sorry that it falls on you as someone who's really trying to solve a different set of problems, but it does, and it should be really easy to do so.

Reply:
I mean, I'd love to have a button that found all the bugs, but I also don't know how to do it. :)

Some sort of static analysis that detected uses of a builder in different arguments of a call would be a good start, but it wouldn't catch indirect uses of the builder, like calling a general emission function on the IGF/SGF.